src/Dropbox.elm

```diff
@@ -861,7 +924,148 @@ upload auth info =
+    | OtherListError String Json.Encode.Value
+    | OtherListFailure Http.Error
+
+
+decodeListError : Json.Decode.Decoder ListFolderError
+decodeListError =
+    Json.Decode.field "error" <|
+        openUnion OtherListError
+            [ tagValue "path" PathListError decodeLookupError
+            , tagVoid "reset" ExpiredCursorError
+            ]
+
+
+{-| See <https://www.dropbox.com/developers/documentation/http/documentation#files-list_folder>
+-}
+type alias ListFolderResponse =
+    { entries : List Metadata
+    , cursor : String
+    , hasMore : Bool
+    }
+
+
+decodeListResponse : Json.Decode.Decoder ListFolderResponse
+decodeListResponse =
+    Pipeline.decode ListFolderResponse
+        |> Pipeline.required "entries"
+            (Json.Decode.list
+                (Json.Decode.oneOf
+                    [ Json.Decode.map FileMeta decodeFileMetadata
+                    , Json.Decode.map FolderMeta decodeFolderMetadata
+                    , Json.Decode.map DeletedMeta decodeDeletedMetadata
+                    ]
+                )
+            )
+        |> Pipeline.required "cursor" Json.Decode.string
+        |> Pipeline.required "has_more" Json.Decode.bool
+
+
+{-| Begin listing the contents of a folder.
+
+See <https://www.dropbox.com/developers/documentation/http/documentation#files-list_folder>
+
+-}
+listFolder : UserAuth -> ListFolderRequest -> Task ListFolderError ListFolderResponse
+listFolder auth options =
+    let
+        url =
+            "https://api.dropboxapi.com/2/files/list_folder"
+
+        body =
+            Json.Encode.encode 0 <|
+                Json.Encode.object <|
+                    [ ( "path", Json.Encode.string options.path )
+                    , ( "recursive", Json.Encode.bool options.recursive )
+                    , ( "include_media_info", Json.Encode.bool options.includeMediaInfo )
+                    , ( "include_deleted", Json.Encode.bool options.includeDeleted )
+                    , ( "include_has_explicit_shared_members", Json.Encode.bool options.includeHasExplicitSharedMembers )
+                    ]
+
+        decodeError err =
+            case err of
+                Http.BadStatus response ->
+                    case Json.Decode.decodeString decodeListError response.body of
+                        Ok err ->
+                            err
+
+                        Err _ ->
+                            OtherListFailure err
+
+                _ ->
+                    OtherListFailure err
+    in
+    Http.request
+        { method = "POST"
+        , headers =
+            [ authHeader auth ]
+        , url = url
+        , body = Http.stringBody "application/json" body
+        , expect = Http.expectJson decodeListResponse
+        , timeout = Nothing
+        , withCredentials = False
+        }
+        |> Http.toTask
+        |> Task.mapError decodeError
+
+
+listFolderContinue : UserAuth -> ListFolderResponse -> Task ListFolderError ListFolderResponse
```

Comment:
In working with this a bit more, I'm thinking taking the `ListFolderResponse` type as the second argument here instead of just the cursor was probably not a good call.